src/MKRWAN.h

```diff
@@ -452,23 +523,18 @@ class LoRaModem : public Stream
     }
     // populate version field on startup
     version();
-    if (!isLatestFW()) {
-      DBG("Please update fw using MKRWANFWUpdate_standalone.ino sketch");
+    if (isArduinoFW() && !isLatestFW()) {
+      DBG("### Please update fw using MKRWANFWUpdate_standalone.ino sketch");
     }
     return true;
   }
 
   bool configureClass(_lora_class _class) {
-    sendAT(GF("+CLASS="), (char)_class);
-    if (waitResponse() != 1) {
-        return false;
-    }
-    return true;
+    return setValue(GF(AT_CLASS), (char)_class);
```

Comment:
Technically, class C is implemented, i.e., the modem stays in listen mode and you have to poll it to see if new data arrived. Didn't try it yet, though. Don't know if I get to use it in the following days. Still working on the test scripts
Class B is not implemented.

Reply:
Class C support requires that we consider that the samd21 might be sleeping between downlinks.  If that is the case, there may be data loss.  Couple ways to avoid it are to ensure SERCOM is properly configured to wake or do not send data until polled.

Reply:
> Class C support requires that we consider that the samd21 might be sleeping between downlinks.
Class C is an always-on mode, thought for line-powered devices. I don't think that MCU sleep will be considered in such application cases. Am I missing out on something here 🤔

Reply:
@flhofer You got a point. Imo when Class C is desired, we can live without sleep modes. However, if we can put the MCU to sleep while the modem keeps doing its job, we could save some energy which is always desirable.

Reply:
@flhofer The polling mechanism which is currently implemented uses Class A. It sends a dummy message to open the RX1 and RX2 receive windows.

Reply:
Every device has to support class A. Actually, the library right now always fills the rx-buffer if it gets something on the `waitresponse` method. However, wouldn't know how that behaves if you don't read the serial for seconds, if not minutes. The library does not support interrupt activation as for example does the TTN uno version. That should be the next thing to work on. This way you can put the MCU to sleep and wake-up on receive. The actual implementation is very basic and unfortunately relies completely on polls.
As an in-between thing you can just periodically read some value (eg. FCD) and then see if there is some data `.available`. Just a suggestion